internal/job/checkout.go

```diff
@@ -217,6 +217,11 @@ func (e *Executor) CheckoutPhase(ctx context.Context) error {
 		}
 	}
 
+	err = e.sendCommitToBuildkite(ctx)
+	if err != nil {
+		e.shell.OptionalWarningf("commit-metadata", err.Error())
```

Comment:
Maybe don't use an error string in the format arg (e.g. imagine if `err.Error()` returns a string with `%s` in it)...

```suggestion
		  e.shell.OptionalWarningf("commit-metadata", "Couldn't %v", err)
```

(Adjust format string or error strings as necessary.)

Reply:
oooh yeah good call. it felt yucky doing it, i'll fix it up

Reply:
?

Reply:
ah shoot, i had this committed locally, but forgot to push it 🤦 i'll make a new PR.

Reply:
https://github.com/buildkite/agent/pull/2699